### DISCUSSION TITLE
Using separate databases for content and configuration

### DISCUSSION THREAD

**Discussion by IvanDrag0** [nocodb/nocodb#2755] [Q&A]
Is it possible to use two different databases for content and configuration (so that the content database remains "clean" of any NocoDB configuration specific items)?

**Comment by wingkwong** [comment]
Please take a look at this - https://docs.nocodb.com/engineering/architecture

Yes. You can create two different databases - one for `NC_DB` which stores the meta data (I think that's what you mean by configuration), another one for content. In this case, you need to create a project by connecting to an external database.

**Reply by IvanDrag0** [reply]
Oh so as long as I set the NC_DB variable in the container (or the OS if it's outside the container), then it will automatically store metadata in a separate database?

**Reply by wingkwong** [reply]
1. If you don't specify `NC_DB` when starting the app, then a sqlite file `noco.db` will be created for you as meta database.
1.1 If you create a project without connecting to external database, all the content and meta in this project will be stored in `noco.db`
1.2 If you create a project by connecting to external database `ext_db`, all the content in this project will be stored in `ext_db` and meta will be stored in `noco.db`.

2. If you specify `NC_DB`, then that database you specified will be used as meta database.
2.1 If you create a project without connecting to external database, all the content and meta in this project will be stored in `NC_DB`
2.2 If you create a project by connecting to external database `ext_db`, all the content in this project will be stored in `ext_db` and meta will be stored in `NC_DB`.
